src/main/java/no/nav/pto/veilarbportefolje/util/JobUtils.java

```diff
@@ -0,0 +1,55 @@
+package no.nav.pto.veilarbportefolje.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.lang.invoke.MethodHandles;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+
+import static java.net.InetAddress.getLocalHost;
+import static no.nav.common.leaderelection.LeaderElection.isNotLeader;
+import static no.nav.common.utils.IdUtils.generateId;
+
+public class JobUtils {
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static String MDC_JOB_ID = "jobId";
+
+    public static Optional<RunningJob> runAsyncJobOnLeader(Runnable runnable) {
+        if (isNotLeader()) {
+            return Optional.empty();
+        }
+        RunningJob runningJob = runAsyncJob(runnable);
+        return Optional.of(runningJob);
+    }
+
+    public static RunningJob runAsyncJob(Runnable runnable) {
+        String jobId = generateId();
+
+        CompletableFuture<Void> future = CompletableFuture.runAsync(
+                () -> {
+                    MDC.put(MDC_JOB_ID, jobId);
+                    runnable.run();
+                    MDC.remove(MDC_JOB_ID);
+                },
+                Executors.newSingleThreadExecutor()
```

Comment:
Her lager den en ny tråd i stede for å plukke en av tre tråder fra ForkJoinPool.commonPool()